File: src/Server/base/receta_diaDesayuno.js
const express = require('express');
const multer = require('multer');
const fs = require('fs');
const path = require('path');
const db = require('./connection');
const router = express.Router();

// Actualizar receta de desayuno
router.put("/desayuno/:id", (req, res) => {
    const { id } = req.params;
    const { fecha, idRecetaDesayuno, idUsuarioModif } = req.body;
  
    if (!fecha || !idRecetaDesayuno || !idUsuarioModif) {
      return res.status(400).json({ message: "Faltan datos requeridos" });
    }
  
    const query = `
      UPDATE recetas_dia
      SET Fecha = ?, Id_Receta_Desayuno = ?, Id_Usuario_Modif = ?, Fecha_Modif = NOW()
      WHERE Id_Recetas_Dia = ?
    `;
  
    db.query(query, [fecha, idRecetaDesayuno, idUsuarioModif, id], (err, result) => {
      if (err) {
        console.error("Error al actualizar desayuno:", err);
        return res.status(500).json({ message: "Error interno del servidor" });
      }
  
      res.json({ message: "Desayuno actualizado con éxito" });
    });
  });
  
// Eliminar (borrar lógico) receta de desayuno
router.delete("/desayuno/:id", (req, res) => {
    const { id } = req.params;
    const { idUsuarioBaja } = req.body;
  
    if (!idUsuarioBaja) {
      return res.status(400).json({ message: "Faltan datos requeridos" });
    }
  
    const query = `
      UPDATE recetas_dia
      SET Id_Receta_Desayuno = NULL, Id_Usuario_Baja = ?, Fecha_Baja = NOW()
      WHERE Id_Recetas_Dia = ?
    `;
  
    db.query(query, [idUsuarioBaja, id], (err, result) => {
      if (err) {
        console.error("Error al eliminar desayuno:", err);
        return res.status(500).json({ message: "Error interno del servidor" });
      }
  
      res.json({ message: "Desayuno eliminado" });
    });
  });
  
  
module.exports = router;

File: src/serverP/base/receta_diaDesayuno.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
import mysql.connector
from .connection import get_db_connection

# Crear el router
recetas_dia_desayuno_router = APIRouter()

# Modelos de validación con Pydantic
class IngredienteSchema(BaseModel):
    id_receta: int
    Id_Usuario_Alta: int

class EliminarSchema(BaseModel):
    Id_Usuario_Modif: int

# Actualizar receta de desayuno
@recetas_dia_desayuno_router.put("/{Id_Recetas_Dia}")
def actualizar_desayuno(Id_Recetas_Dia: int, data: IngredienteSchema):
    query = """
        UPDATE recetas_dia
        SET Id_Receta_Desayuno = %s, Id_Usuario_Modif = %s, Fecha_Modif = NOW()
        WHERE Id_Recetas_Dia = %s
    """

    try:
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute(query, (data.id_receta, data.Id_Usuario_Alta, Id_Recetas_Dia))
        conn.commit()
        return {"message": "Desayuno actualizado con éxito"}
    except mysql.connector.Error as err:
        raise HTTPException(status_code=500, detail={"message": "Error interno del servidor", "error": str(err)})
    finally:
        cursor.close()
        conn.close()

# Eliminar (borrado lógico) receta de desayuno
@recetas_dia_desayuno_router.put("/borrar/{Id_Recetas_Dia}")
def eliminar_desayuno(Id_Recetas_Dia: int, data: EliminarSchema):
    query = """
        UPDATE recetas_dia
        SET Id_Receta_Desayuno = NULL, Id_Usuario_Modif = %s, Fecha_Modif = NOW()
        WHERE Id_Recetas_Dia = %s
    """

    try:
        conn = get_db_connection()
        cursor = conn.cursor()
        cursor.execute(query, (data.Id_Usuario_Modif, Id_Recetas_Dia))
        conn.commit()
        return {"message": "Desayuno eliminado"}
    except mysql.connector.Error as err:
        raise HTTPException(status_code=500, detail={"message": "Error interno del servidor", "error": str(err)})
    finally:
        cursor.close()
        conn.close()
